agent/agent.go

```diff
@@ -360,6 +361,7 @@ func (a *Agent) startLogging(ctx context.Context, tc *taskContext) error {
 		tc.logger, err = a.makeLoggerProducer(ctx, tc, &model.LoggerConfig{}, "")
 	}
 	if err != nil {
+		grip.Alert(errors.Wrap(err, "problem making the logger producer"))
```

Comment:
It seems like logging to Splunk directly is not the right solution here, except in extreme cases.

It seems like there are alternative solutions to this. If we know that `startLogging` must come before the task actually starts executing, I think it would be better if dealing with logging setup failure was done on the server side (e.g. in`EndTask`) rather than the agent side. I would just add a FailureReason string or something to TaskEndDetail indicating the reason the task ended in failure and you can check against that failure for special handling on the server side.

Reply:
Are you suggesting adding more fields to the `TaskEndDetail` struct?

Reply:
Yeah, I would add either an error code like an int or a hard-coded string that can be understood on both sides as meaning "the logger failed to set up". That way, the app server can decide how it deals with exceptional error cases in tasks separately from the agent.

Reply:
The issue is that I want to see why it failed. And error logging that only logs it to the host's filesystem, which is a problem since the majority of our hosts are ephemeral and terminate quickly. In other words, I need more persistent logging. I would also argue that when the logging fails to setup its pretty severe and should really rarely happen. Cedar is having issues with scaling right now, which is what I think is the root cause of these initial task failures but I can't be sure until I have better information.

Reply:
If it's a temporary log, then I guess it's alright as long as it gets removed later.